ChessGame/static/chessgame/js/ChessPieces/rule.js

```diff
@@ -0,0 +1,11 @@
+class Rule {
+  canCapture;
+  rowOp;
+  columnOp;
+
+  constructor(canCapture, rowOp, columnOp) {
+    this.canCapture = canCapture;
```

Comment:
This is a really interesting way to encode the chess piece rules! I'm really curious to see how this works out. I have a feeling this may be hard to work with as you build out the game logic but I'd definitely keep exploring this.

There's always the option of working backwards from where someone's trying to move (figure out if space is invalid or not as they try to move) or even simpler, every piece on the board keeps a 2D array of bools that contain all of it's currently valid moves. Then you don't have to go with this delta approach.

Reply:
Okay true that might be an easier way - I was going to check based on which piece was clicked and then highlight the available spaces

Reply:
2D array approach should be easy to work with in the long run. You'd update every pieces grid of available moves every time a piece moves essentially.

Reply:
Okay I agree I'll switch to that